source/common/http/conn_manager_config.h

```diff
@@ -500,6 +500,11 @@ class ConnectionManagerConfig {
    * @return maximum requests for downstream.
    */
   virtual uint64_t maxRequestsPerConnection() const PURE;
+  /**
+   * @return the config describing if/how to write the Proxy-Status HTTP response header.
+   * If nullptr, don't write the Proxy-Status HTTP response header.
+   */
+  virtual const HttpConnectionManagerProto::ProxyStatusConfig* proxyStatusConfig() const PURE;
```

Comment:
nit: the only small suggestion I have is to add `nameForProxyStatus()` method to the config and set it to either node ID (you can get it from the `context` constructor parameter) or the envoy literal. In this case you do not need to plumb the node id into the `mutateResponseHeaders` and can just get it from the config.

This is not a big deal, but might be helpful if we later need to be able to specify proxy name in the proxy status config.